### PR TITLE
Added mobile flat route handling

### PR DESCRIPTION
**Pull request by ChristianMoll**
@plentymarkets/team-terra

### CODE REVIEW

**Comment by dantrau** [comment]
👍 tested

File: src/app/components/layouts/column-container/terra-two-columns-container.component.ts [hunk 1]
```diff
@@ -37,11 +54,55 @@ export class TerraTwoColumnsContainerComponent
...
     {
+        this.leftColumn = 'hidden-xs ' + this.leftColMD() + this.leftColLG();
+        this.rightColumn = this.leftRightColXS() + this.rightColMD() + this.rightColLG();
+
+        let subscribable:Observable<Event> = this.router.events.filter((event:RouterEvent) =>
+        {
+            return event instanceof NavigationEnd;
+        });
+
+        this.subscription = subscribable.subscribe((event:NavigationEnd) =>
```
Comment (MFrank84) [inline src/app/components/layouts/column-container/terra-two-columns-container.component.ts]: Is this necessary in the constructor or can it be moved to `ngOnInit()`? Please keep the constructor as short as possible.

File: src/app/components/layouts/column-container/terra-two-columns-container.component.ts [hunk 2]
```diff
@@ -37,11 +54,55 @@ export class TerraTwoColumnsContainerComponent
...
+            }
+        });
+
         this.leftColumnWidth = this._leftColumnWidth; // trigger calculation for default values
     }
 
+    public ngOnInit():void
+    {
+        this.leftColumn = 'hidden-xs ' + this.leftColMD() + this.leftColLG();
+        this.rightColumn = this.leftRightColXS() + this.rightColMD() + this.rightColLG();
```
[the remark this answers is not in the document]
Comment (MFrank84) [reply src/app/components/layouts/column-container/terra-two-columns-container.component.ts]: Please add a `'hidden-xs'` as a readonly attribute. Then you can create a `hidden-xs()` function that returns `this.hiddenXs + this.spacer. Analogous to the other functions

File: src/app/components/layouts/column-container/terra-two-columns-container.component.ts [hunk 3]
```diff
@@ -37,11 +53,61 @@ export class TerraTwoColumnsContainerComponent
...
+
+        let subscribable:Observable<Event> = this.router.events.filter((event:RouterEvent) =>
+        {
+            return event instanceof NavigationEnd && event.urlAfterRedirects.startsWith(this.basePath);
+        });
+
+        this.subscription = subscribable.subscribe((event:NavigationEnd) =>
+        {
+            this.leftColumn = this.leftRightColXS() + this.leftColMD() + this.leftColLG();
+            this.rightColumn = this.leftRightColXS() + this.rightColMD() + this.rightColLG();
```
Comment (pweyrich) [inline src/app/components/layouts/column-container/terra-two-columns-container.component.ts]: Das muss hier eigentlich nicht ausgeführt werden, das wird ja anschließend in `this.setColumnHidden()` aktualisiert, oder?

File: src/app/components/layouts/column-container/terra-two-columns-container.component.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+@media (max-width: 767px)
```
Comment (pweyrich) [inline src/app/components/layouts/column-container/terra-two-columns-container.component.scss]: isn't there a scss variable that can be used for the breakpoint?

File: src/app/components/layouts/column-container/terra-two-columns-container.component.ts [hunk 5]
```diff
@@ -37,11 +53,61 @@ export class TerraTwoColumnsContainerComponent
...
+                this.setColumnHidden('left');
+            }
+        });
+
         this.leftColumnWidth = this._leftColumnWidth; // trigger calculation for default values
     }
 
+    public ngOnInit():void
+    {
+        this.setColumnHidden('left');
```
Comment (pweyrich) [inline src/app/components/layouts/column-container/terra-two-columns-container.component.ts]: call this function either in the `constructor` or in `ngOnInit`, not both.

File: src/app/components/layouts/column-container/terra-two-columns-container.component.scss (hunk 4 above)
Comment (ChristianMoll) [reply src/app/components/layouts/column-container/terra-two-columns-container.component.scss]: not in components by the look of it

File: src/app/components/layouts/column-container/terra-two-columns-container.component.ts (hunk 5 above)
Comment (ChristianMoll) [reply src/app/components/layouts/column-container/terra-two-columns-container.component.ts]: done

**Comment by ChristianMoll** [comment]
https://github.com/plentymarkets/terra-components/pull/585

**Pull request closed without merging**